pkg/git/gitlab/gitlab_client.go

```diff
@@ -53,6 +54,13 @@ func (g *GitlabClient) EnsurePaCMergeRequest(repoUrl string, d *gp.MergeRequestD
 		}
 		d.BaseBranchName = baseBranch
 	}
+	baseBranchExists, err := g.branchExist(projectPath, d.BaseBranchName)
```

Comment:
I think it make sense to check for branch existance only if `BaseBranchName` was set.

Reply:
yes